dual_net.py

```diff
@@ -533,8 +533,16 @@ def freeze_graph(model_path):
 def freeze_graph_tpu(model_path):
     """Custom freeze_graph implementation for Cloud TPU."""
 
+    assert model_path
     assert FLAGS.tpu_name
-    sess = tf.Session(FLAGS.tpu_name)
+    if FLAGS.tpu_name.startswith('grpc://'):
+        sess = tf.Session(FLAGS.tpu_name)
+    else:
+        tpu_cluster_resolver = tf.contrib.cluster_resolver.TPUClusterResolver(
+            FLAGS.tpu_name, zone=None, project=None)
+        tpu_grpc_url = tpu_cluster_resolver.get_master()
+        sess = tf.Session(tpu_grpc_url)
```

Comment:
can deduplicate this line by moving `sess = tf.Session(tpu_grpc_url)` out of the if/else, and setting `tpu_grpc_url = FLAGS.tpu_name` in the True branch.

Reply:
Done.